demonstrations/tutorial_qgrnn.py

```diff
@@ -0,0 +1,609 @@
+                m2 = np.kron(m2, qml.PauliX.matrix)
+            else:
+                m1 = np.kron(m1, np.identity(2))
+                m2 = np.kron(m2, np.identity(2))
+        matrix = np.add(matrix, np.add(params[1][i] * m1, m2))
+
+    return matrix
+
+# Prints a visual representation of the Hamiltonian matrix
+ham_matrix = create_hamiltonian_matrix(qubit_number, ising_graph, matrix_params)
+plt.matshow(ham_matrix)
+plt.show()
+
+
+######################################################################
+# Preparing Quantum Data
+# ^^^^^^^^^^^^^^^^^^^^^^
+#
+
+
+######################################################################
+# The collection of quantum data needed to run the QGRNN has two componenets: 
+# copies of a low-energy state, and a collection of time-evolved state, each of which are
+# simply the low-energy state evolved to different times. 
+# For the target Hamiltonian we 
+# defined, we find that a low-energy state is given by the following vector
+# (represented by a Numpy array):
+#
+
+low_energy_state = np.array([-0.02086666+0.00920016j, -0.00379192-0.00859852j,  0.06594626-0.02907913j,
+  0.0119852 +0.02717445j,  0.07633593-0.03366391j,  0.01387486+0.03145572j,
+ -0.24124938+0.10640219j, -0.04385454-0.09941154j,  0.06397641-0.02820407j,
+  0.01162454+0.02636273j, -0.20218878+0.08914518j, -0.03674192-0.08331586j,
+ -0.23404312+0.10320031j, -0.04253486-0.09644206j,  0.73966164-0.32618732j,
+  0.13444079+0.30479209j])
+
+
+######################################################################
+# We can verify that this is a low-energy
+# state by numerically finding the lowest eigenvalue of the Hamiltonian
+# matrix and comparing it to the energy expectation of the low-energy state:
+#
+
+# Finds the energy expectation
+
+def expectation_value(vector, matrix):
+    return np.inner(np.conj(vector), (matrix @ vector))
+
+energy_exp = expectation_value(low_energy_state, ham_matrix)
+print("Energy Expectation: {}".format(energy_exp))
+
+def ground_state_energy(matrix):
+
+    # Finds the eigenstates of the matrix
+    val = np.linalg.eig(matrix)[0]
+
+    # Returns the minimum eigenvalue
+    return min(val)
+
+ground_state = ground_state_energy(ham_matrix)
+print("Ground State Energy: {}".format(ground_state))
+
+
+######################################################################
+# This shows us that we have in fact found a low-energy, non-ground state, 
+# as the energy expectation is slightly greater than the energy of the true ground
+# state. This, however, is only half of the information we need. We also require 
+# a collection of time-evolved states. 
+# Luckily, evolving a state forward in time is fairly straightforward, all we 
+# have to do is multiply the initial state by a time-evolution unitary. We define 
+# this operation as a custom gate in PennyLane:
+#
+
+
+def state_evolve(hamiltonian, qubits, time):
+
+    U = scipy.linalg.expm(complex(0, -1) * hamiltonian * time)
```

Comment:
```suggestion
    U = scipy.linalg.expm(-1j* hamiltonian * time)
```
`1j` is square-root of -1 in python